### PR TITLE
Fix incorrect data filtering

### PR DESCRIPTION
**Pull request by jpwhite4**
Fixes issue with data filtering when an unpriv user tries to view
timeseries data for a realm / stat that is protected (i.e. has
additional role-specific filters applied).

Added integration test that confirms the correct behaviour.

### CODE REVIEW

File: classes/DataWarehouse/Query/Query.php [hunk 1]
```diff
@@ -756,6 +756,21 @@ public function setParameters(array $parameters = array())
         }
     }
 
+    /**
+     * Copy the parameters and role parameters from another query class
+     * The where conditions and role parameters from the other class will
+     * overwrite any existing settings in this class.
+     */
+    public function cloneParameters($other)
```
Comment (smgallo) [inline classes/DataWarehouse/Query/Query.php]: We should use `(Query $other)` in the definition if we are always expecting an object of that class.

**Comment by smgallo** [comment]
Other than the comment above, LGTM.

**Comment by jpwhite4** [comment]
I note that the shippable build failed because the expected results were changed as a consequence of a different pull request that has not yet been approved. We will have to come up with a better way of dealing with external test artefact dependencies.

**Comment by ryanrath** [comment]
That's my bad, I've reverted the test artifact so that your test should pass now and created a new artifact with the new expected test results for my branch ( and the test that uses it ).

**Pull request merged**